lms/djangoapps/courseware/model_data.py

```diff
@@ -840,6 +840,8 @@ def set_many(self, kv_dict):
             kv_dict (dict): dict mapping from `DjangoKeyValueStore.Key`s to field values
         Raises: DatabaseError if any fields fail to save
         """
+        if self.read_only:
```

Comment:
In terms of features that assume a field is present -- I've done spot checks of a bunch of our common block types. It's possible that some functionality will break somewhere, but I think that the overall risk is low, especially since this should only affect crawlers.